### ISSUE TITLE
Unable to stop current pledge

### ISSUE THREAD

**Issue by QuincyLarson** [FreeCodeCamp/FreeCodeCamp#8794]
Campers used to be able to delete pledges, but this important functionality somehow recently broke.

Steps to reproduce:
1. Go to https://www.freecodecamp.com/commit and create a pledge
2. Return to https://www.freecodecamp.com/commit and attempt to end your pledge by clicking "stop my current pledge"
You will get an error.

![](https://www.evernote.com/l/Ar-Ni4Q38ypFL7varpW4Ma_ujSSVMqGcyQoB/image.png)

~~The way we are fixing this manually is by adding a "formerUserId" property set to the value of "userId", and then setting "userId" itself null.~~

edit(@berkeleyTrue): removed extra information as it just confuses the issue

**Comment by gasperno** [comment]
I need a clarification regarding the stop-commit functionality. I am consolidating the two steps. Correct me if I am wrong.
Step 1: When I hit stop my current pledge, the database table modifies a particular document in 'pledge' collection. Modified fields would be formerUserId and [few other fields](https://github.com/FreeCodeCamp/FreeCodeCamp/blob/staging/server/boot/commit.js#L212).
and then throw a flash message that displays

Step 2:
+ You have successfully unpledged and flash message comes after redirect to freecodecamp/<username>
+ Error message/ Failure Message and flash message comes after redirect(? or not) to freecodecamp/<username>

**Comment by gasperno** [comment]
@QuincyLarson, While debugging I noticed that the modification of fields is happening, but flash message logic seemed faulty to me. 
I was little confused when I saw your comment 

> The way we are **fixing this manually** 

I was confused if I understood the functionality correctly.

**Comment by BerkeleyTrue** [comment]
@gasperno please ignore that bit of information. The issue is why is it throwing an error and why is it not correctly modifying the pledge object.

**Comment by gasperno** [comment]
It throws a flash with error background. But the object modification is as expected.
Before hitting Stop my current pledge,
```
{
	"_id" : ObjectId("57470ed82e6d33f57713b16c"),
	"amount" : 10,
	"dateStarted" : ISODate("2016-05-26T14:57:28.200Z"),
	"isCompleted" : "false",
	"userId" : ObjectId("572bd5106469e0f1dd******”),
	"displayName" : "Black Girls CODE"
}
```
After hitting the button to stop my pledge,
```
{
	"_id" : ObjectId("57470ed82e6d33f57713b16c"),
	"amount" : 10,
	"dateStarted" : ISODate("2016-05-26T14:57:28.200Z"),
	"dateEnded" : ISODate("2016-05-26T14:58:50.463Z"),
	"formerUserId" : "572bd5106469e0f1dd*****",
	"isOrphaned" : true,
	"isCompleted" : "false",
	"userId" : null,
	"goal" : "Front End Development Certification",
	"displayName" : "Black Girls CODE",
}
```
Note: I ignored some fields that are unmodified.
That clearly implies that the object modification is taking place. **Is this the expected modification in mongo collection?**
However it throws a flash component as error because of [the few lines beginning here] (https://github.com/FreeCodeCamp/FreeCodeCamp/blob/staging/server/boot/commit.js#L220).

Fix: req.flash('success', "You have successfully unpledged") and req.flash("errors","<message>"); in commit.js after stopCommit. 
I will try to submit a PR before tonight.

**Comment by BerkeleyTrue** [comment]
I don't think you've gotten to the core of the issue. You did find a bug, but not the one responsible here. The error we are seeing is an 'oops' error, meaning an actual error was thrown in production and caught by the server, causing the oops error to be shown. 

In development the server will actually print out the error stack as a page when that happens.

**Comment by gasperno** [comment]
![screen shot 2016-05-26 at 5 40 44 pm](https://cloud.githubusercontent.com/assets/7889179/15591083/0e938154-2369-11e6-81e1-2196f05cbc78.png)
I din't check it on freecodecamp until now. I was checking on my local. I attached the screenshot of the local machine screen after I stop my pledge. You are right. I dint get Oops error when I ran on my local.

**Comment by gasperno** [comment]
My code was not updated. Sorry for the confusion.

**Comment by tlerias** [comment]
This is my first Open Source contribution! I fixed this bug in PR #8824
Feedback is appreciated.
